src/meta/src/manager/catalog/mod.rs

```diff
@@ -1266,13 +1284,22 @@ where
         } else {
             database_core.mark_creating(&key);
             user_core.increase_ref(source.owner);
+            if let Some(connection_id) = source.connection_id {
```

Comment:
Need to ensure existence of the connection id, as the connection could be dropped from other frontend at the same time.

Reply:
wow! I didnt think of that! so considerate!